caas/kubernetes/provider/cloud.go

```diff
@@ -125,31 +130,45 @@ func UpdateKubeCloudWithStorage(k8sCloud *cloud.Cloud, credential cloud.Credenti
 	// If the user has not specified storage, check that the cluster has Juju's opinionated defaults.
 	cloudType := cloud.SplitHostCloudRegion(storageParams.HostCloudRegion)[0]
 	err = storageParams.MetadataChecker.CheckDefaultWorkloadStorage(cloudType, clusterMetadata.NominatedStorageClass)
-	if errors.IsNotFound(err) {
-		return fail(UnknownClusterError{CloudName: cloudType})
-	}
-	if storageParams.WorkloadStorage == "" && caas.IsNonPreferredStorageError(err) {
-		npse := err.(*caas.NonPreferredStorageError)
-		return fail(NoRecommendedStorageError{Message: err.Error(), ProviderName: npse.Name})
-	}
-	if err != nil && !caas.IsNonPreferredStorageError(err) {
-		return fail(errors.Trace(err))
-	}
 
-	// If no storage class exists, we need to create one with the opinionated defaults.
+	if storageParams.WorkloadStorage == "" {
+		if errors.IsNotFound(err) {
+			return fail(UnknownClusterError{CloudName: cloudType})
+		}
+		if caas.IsNonPreferredStorageError(err) {
+			npse := err.(*caas.NonPreferredStorageError)
+			return fail(NoRecommendedStorageError{Message: err.Error(), ProviderName: npse.Name})
+		}
+		if err != nil {
+			return fail(errors.Trace(err))
+		}
+	}
```

Comment:
do we want to return any known errors for `storageParams.WorkloadStorage != ""` case?

Reply:
workload storage != "" is handled below